feat(logo): add support for Material icons and emojis in st.logo

rahuld109 opened the pull request:
## Describe your changes

Adds support for Material icons and emojis in `st.logo()`:

```python
st.logo(":material/home:")
st.logo("🏠")
st.logo(":material/rocket_launch:", link="https://streamlit.io")
st.logo(full_logo, icon_image=":material/menu:")
```

Following @lukasmasuch's suggestion, this passes the icon name to the frontend and reuses the existing `DynamicIcon` component (same pattern `st.chat_message` uses for avatars).

Changes:
- New `ImageType` enum in Logo protobuf (IMAGE/EMOJI/ICON)
- `_process_logo_image()` helper to detect input type
- Frontend `LogoComponent` renders `DynamicIcon` for icons/emojis
- New `StyledIconLogo` styled component for sizing

## Screenshots

### Material Icons - Size Parameter (small / medium / large)

| Small | Medium | Large |
|-------|--------|-------|
| \![small](https://raw.githubusercontent.com/rahuld109/streamlit/temp-screenshots/.github/screenshots/material_rocket_small_expanded.png) | \![medium](https://raw.githubusercontent.com/rahuld109/streamlit/temp-screenshots/.github/screenshots/material_rocket_medium_expanded.png) | \![large](https://raw.githubusercontent.com/rahuld109/streamlit/temp-screenshots/.github/screenshots/material_rocket_large_expanded.png) |

### Different Material Icons

| :material/home: | :material/favorite: |
|-----------------|---------------------|
| \![home](https://raw.githubusercontent.com/rahuld109/streamlit/temp-screenshots/.github/screenshots/material_home_medium_expanded.png) | \![favorite](https://raw.githubusercontent.com/rahuld109/streamlit/temp-screenshots/.github/screenshots/material_favorite_medium_expanded.png) |

### Emoji Support

\![emoji](https://raw.githubusercontent.com/rahuld109/streamlit/temp-screenshots/.github/screenshots/emoji_rocket_small_expanded.png)

## GitHub Issue Link (if applicable)

Closes #13385

## Testing Plan

- Unit Tests (JS and/or Python)
  - Added 5 Python tests for icons, emojis, icon with link, icon_image as icon, and invalid icon
  - Added 4 frontend tests for DynamicIcon rendering with ICON/EMOJI types
- E2E Tests: Not added - relies on existing `DynamicIcon` E2E coverage
- Manual testing: Verified icons and emojis render correctly in sidebar

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

rahuld109 committed:
feat(logo): add support for Material icons and emojis in st.logo

Add support for using Material icons (e.g., `:material/home:`) and
emojis (e.g., `🏠`) as logos in `st.logo()`, in addition to image files.

Changes:
- Add ImageType enum to Logo protobuf (IMAGE, EMOJI, ICON)
- Add _process_logo_image() helper to detect and process icon/emoji/image
- Integrate DynamicIcon component in frontend LogoComponent
- Add StyledIconLogo styled component for icon/emoji rendering
- Update docstrings to document new supported formats

Closes #13385

lukasmasuch commented:
@cursor review

rahuld109 committed:
fix: address review feedback - map icon size to logo size

- Map logo size (small/medium/large) to DynamicIcon size (lg/xl/twoXL)
- Fix type annotation for image parameter to accept str for emoji/icons

rahuld109 commented on frontend/app/src/components/Logo/LogoComponent.tsx:
```diff
@@ -96,7 +100,20 @@ const LogoComponent = ({
+      className="stLogo"
+      data-testid={dataTestId}
+    >
+      <DynamicIcon iconValue={displayImage} size="xl" />
```
Fixed! Added a mapping from logo size to icon size:
- small → lg (1.25rem)
- medium → xl (1.5rem)
- large → twoXL (1.8rem)

rahuld109 commented on lib/streamlit/commands/logo.py:
```diff
@@ -35,6 +37,46 @@ def _invalid_logo_text(field_name: str) -> str:
+
 @gather_metrics("logo")
 def logo(
     image: AtomicImage,
```
Fixed! Updated the type annotation to `AtomicImage | str` for both `image` and `icon_image` parameters.

rahuld109 committed:
Merge branch 'develop' into feat/logo-icon-support

rahuld109 committed:
Merge branch 'develop' into feat/logo-icon-support

rahuld109 committed:
Merge branch 'develop' into feat/logo-icon-support

jrieke commented:
Can you please add some screenshots of what this looks like with different emojis and icons, and different values for the `size` parameter of `st.logo`?

rahuld109 committed:
Merge branch 'develop' into feat/logo-icon-support

rahuld109 committed:
Merge branch 'develop' into feat/logo-icon-support

rahuld109 committed:
Merge branch 'develop' into feat/logo-icon-support

rahuld109 committed:
Fix IconSize import to use internal path alias

rahuld109 committed:
Add edge case tests for st.logo icon/emoji support

rahuld109 committed:
Add E2E tests for material icon and emoji logos

rahuld109 commented:
Hey @jrieke @lukasmasuch, apologies for going quiet on this one - life got in the way for a bit!

I've made all the updates:

- **Screenshots**: Added to the PR description showing material icons at different sizes and emoji support
- **Edge case tests**: Covered invalid icons, empty strings, and invalid input handling
- **E2E tests**: Added tests for both material icon and emoji logo rendering

Let me know if there's anything else - always happy to help!

rahuld109 committed:
Fix ruff format for test_emoji_logo function signature

rahuld109 committed:
Fix IconSize import and add E2E snapshots for icon/emoji logos

rahuld109 commented:
Checks are passing now.

rahuld109 commented:
@jrieke please can you remove `missing:visual-demonstration` label or I need to add any more visuals documentation. Thanks

jrieke commented:
Sorry for the delay. Reviewed the screenshots and they look good, approved from product side! Will send this to our eng team to review.

lukasmasuch committed:
Merge branch 'develop' into pr/rahuld109/13416

lukasmasuch committed:
Update

lukasmasuch reviewed (approved):
LGTM 👍 Thanks for the contribution.

jrieke commented:
Awesome, thanks for the contribution! @rahuld109 would love to send you some swag as a little thank you, feel free to fill out this form: https://docs.google.com/forms/d/e/1FAIpQLSe7cXh3H1DmrgNpVew9qViGIHX7vdEbTv5isA44_z5bgaKTKg/viewform
